Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,104 @@
+# Creates a pull request to boneyard a formula.
+#
+# Usage: brew boneyard-formula-pr [options...] <formula-name>
+#
+# Options:
+#   --dry-run:  Print what would be done rather than doing it.
+
+require "formula"
+require "utils/json"
+require "fileutils"
+
+module Homebrew
+  def boneyard_formula_pr
+    formula = ARGV.formulae.first
+    odie "No formula found!" unless formula
+
+    formula_path = formula.path
+    core = formula_path.to_s.include? "homebrew-core"
+    if ARGV.dry_run?
+      ohai "brew update"
```

Comment:
Actually, this dry-run is a lot of duplicated code to keep in sync. Could it just define local functions like `x_safe_system` and `x_cp` that either do their action or emit a description instead, depending on whether `--dry-run` is on? Easier to read, and then you'd be more sure the dry run reflected what would happen in the live run.